int32 x
int32 y
float32 theta
int8 length
int8 width
string color